### PR TITLE
feat(deb_remote): add sync option parameters

### PR DESCRIPTION
**Pull request by joey-grant**
Changeset adds parameter options and tests for:
- sync_installer
- sync_sources
- sync_udebs

### CODE REVIEW

**Comment by joey-grant** [comment]
This PR addresses https://github.com/pulp/squeezer/issues/152

File: tests/playbooks/deb_remote.yaml [hunk 1]
```diff
@@ -88,6 +88,75 @@
...
+      pulp.squeezer.deb_remote:
+        name: test_deb_remote
+        sync_installer: true
+        state: present
+      register: result
+    - name: Verify modify sync_installer
+      assert:
+        that:
+          - result.changed == true
+          - result.remote.sync_installer == true
```
Comment (mdellweg) [inline tests/playbooks/deb_remote.yaml]: For the sake of brevity (and testing resources...) can you move all three options into the same two module invocations?

**Review by mdellweg — COMMENTED** [review]
Thank you for this contribution.
In order to get this merged, you will need to record the fixtures for the tests. Please take a look at:
https://github.com/pulp/squeezer?tab=readme-ov-file#running-tests-against-a-pulp-in-one-container

File: tests/playbooks/deb_remote.yaml (hunk 1 above)
Comment (joey-grant) [reply tests/playbooks/deb_remote.yaml]: @mdellweg, I combined the tests and added the fixture files. I've squashed the commits and added a `fixes #152` to the bottom of the commit messages. Let me know if there is anything out of order here.

**Comment by mdellweg** [comment]
I'm trying to fix the CI issues here: https://github.com/pulp/squeezer/pull/174

File: plugins/modules/deb_remote.py [hunk 2]
```diff
@@ -31,6 +31,22 @@
       - immediate
       - on_demand
       - streamed
+  sync_installer:
+    description:
+      - Sync installer packages
+    type: boolean
```
Comment (mdellweg) [inline plugins/modules/deb_remote.py]: looks like this needs to be `bool`

Comment (joey-grant) [reply plugins/modules/deb_remote.py]: Sorry for overlooking that. I've pushed the change. Let me know if there is anything else.

**Review by mdellweg — APPROVED** [review]

**Pull request merged**